### PR TITLE
Text Handling Version 2

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-279

#### Description

This PR's main focus is to make the text handling in dpl-cms compatible with the new version in dpl-react (https://github.com/danskernesdigitalebibliotek/dpl-react/pull/212)

What is being done:
* dpl_react_render detects if an application property is an array and converts the value to json if so. This is needed for more complex text values like plural definitions.
* changed the strings that has been changed in order to use the new format in dpl-react
* Adding strings from the serach related apps dpl-react that has not been added before.

#### Checklist

- [x] My complies with [our coding guidelines](../documentation/code-guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

When https://github.com/danskernesdigitalebibliotek/dpl-react/pull/212 is merged, the reference to the dpl-react package will change in composer.json

### CODE REVIEW

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php [hunk 1]
```diff
@@ -169,11 +191,24 @@ public function work(string $wid): array {
...
+        'find-on-shelf-modal-list-material-text' => $this->t('Material', [], $c),
+        'find-on-shelf-modal-list-material-text' => $this->t('Material', [], $c),
+        'find-on-shelf-modal-no-location-specified-text' => $this->t('-', [], $c),
+        'find-on-shelf-modal-no-location-specified-text' => $this->t('No location for find on shelf specified', [], $c),
+        'find-on-shelf-modal-periodical-edition-dropdown-text' => $this->t('Find on shelf modal periodical dropdown - choose edition/volume', [], $c),
+        'find-on-shelf-modal-periodical-edition-dropdown-text' => $this->t('Find on shelf modal periodical dropdown - choose edition/volume', [], $c),
+        'find-on-shelf-modal-periodical-year-dropdown-text' => $this->t('Choose periodical year', [], $c),
+        'find-on-shelf-modal-periodical-year-dropdown-text' => $this->t('Find on shelf modal periodical dropdown - choose year', [], $c),
         'find-on-shelf-modal-screen-reader-modal-description-text' => $this->t('Modal for reservation', [], $c),
+        'find-on-shelf-modal-screen-reader-modal-description-text' => $this->t('Reservation modal screen reader description', [], $c),
```
Comment (kasperg) [inline web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:211]: Please rework the duplicated prop here.

**Review by kasperg — CHANGES_REQUESTED** [review]
Please read into the PHPStan comments

I wonder why it has not reported on the duplicate array entries in all the changesets.

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php (hunk 1 above)
Comment (spaceo) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:211]: 👍

File: composer.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
             "type": "package",
             "package": {
                 "name": "danskernesdigitalebibliotek/dpl-react",
-                "version": "0.3.0-rc8",
+                "version": "0.3.0-rc9",
                 "type": "drupal-library",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/release-release%2F4/dist.zip",
+                    "url": "https://github.com/reload/dpl-react/releases/download/release-handtering-af-placeholders-og-flertal-i-tekstrenge-DDFSOEG-279/dist.zip",
```
[the remark this answers is not in the document]
Comment (spaceo) [reply composer.json:23]: 👍

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php [hunk 3]
```diff
@@ -183,6 +218,9 @@ public function work(string $wid): array {
         'identifier-text' => $this->t('Identifiers', [], $c),
         'in-same-series-text' => $this->t('In the same series', [], $c),
         'in-series-text' => $this->t('in the series', [], $c),
+        'infomedia-modal-close-modal-aria-label-text' => $this->t('Close infomedia modal', [], $c),
+        'infomedia-modal-screen-reader-modal-description-text' => $this->t('Infomedia modal screen reader description', [], $c),
+        'infomedia-modal-screen-reader-modal-description-text' => $this->t('Modal for infomedia', [], $c),
```
[the remark this answers is not in the document]
Comment (spaceo) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php]: 👍

**Review by kasperg — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Pull request merged**